### PR TITLE
chore(python): Bump python min to 3.9 [all tests ci]

### PR DESCRIPTION
**Pull request by lsetiawan**
Bumping python requirement to >=3.9 following xarray.

Ref: #1144

### CODE REVIEW

**Comment by lsetiawan** [comment]
This will fail since PR https://github.com/OSOceanAcoustics/echopype/pull/1145 is not merged yet.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10"]
         runs-on: [ubuntu-latest]
         experimental: [false]
         # TODO: Uncomment when netcdf is unpinned
```
Comment (emiliom) [inline .github/workflows/build.yaml:27]: This is unrelated to this PR per say, but is this TODO something we should act on now that netcdf is unpinned?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10"]
         runs-on: [ubuntu-latest]
         experimental: [false]
         # TODO: Uncomment when netcdf is unpinned
```
Comment (emiliom) [inline .github/workflows/pr.yaml:23]: This is unrelated to this PR per say, but is this TODO something we should act on now that netcdf is unpinned?

**Review by emiliom — APPROVED** [review]
Looks good to me (assuming the CI runs successfully). Thanks!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (lsetiawan) [reply .github/workflows/build.yaml:27]: Thanks for catching that! I'll add to a separate PR

**Comment by lsetiawan** [comment]
Currently CI will most likely fail: https://github.com/OSOceanAcoustics/echopype/pull/1146#issuecomment-1701575398

**Pull request merged**